### PR TITLE
[Bug][Secretariat] Problème de modification du RDV

### PR DESCRIPTION
**Pull request by jjf21**
In the `agents` collection, only agents with same service `rdv.motif.service` were matched.
Leaving agents from `secretariat` even if `rdv.motif.for_secretariat == true`

### CODE REVIEW

File: app/models/agent.rb [hunk 1]
```diff
@@ -26,6 +26,9 @@ class Agent < ApplicationRecord
   scope :active, -> { where(deleted_at: nil) }
   scope :order_by_last_name, -> { order(Arel.sql('LOWER(last_name)')) }
   scope :secretariat, -> { joins(:service).where(services: { name: 'Secrétariat'.freeze }) }
+  scope :available_for_rdv, lambda { |motif, service|
+    motif.for_secretariat ? joins(:service).where(service: service).or(secretariat) : where(service: service)
```
Comment (adipasquale) [inline app/models/agent.rb:30]: I don't think the joins is necessary in the first part of the condition, is it?

File: app/models/agent.rb [hunk 2]
```diff
@@ -26,6 +26,9 @@ class Agent < ApplicationRecord
   scope :active, -> { where(deleted_at: nil) }
   scope :order_by_last_name, -> { order(Arel.sql('LOWER(last_name)')) }
   scope :secretariat, -> { joins(:service).where(services: { name: 'Secrétariat'.freeze }) }
+  scope :available_for_rdv, lambda { |motif, service|
```
Comment (adipasquale) [inline app/models/agent.rb:29]: also in my mind `available_for_rdv` refers to being available on a specific time for a specific rdv, whereas here it's a more generic , someone can do that motif.

I'd be happier with something like `can_perform_motif`, wdyt?

Comment (adipasquale) [inline app/models/agent.rb:29]: by the way, no need to pass both the motif and the service as parameters as it's the service's motif that's used

File: app/models/agent.rb (hunk 1 above)
Comment (jjf21) [reply app/models/agent.rb:30]: without I can't do .or() because both parts must be identical

Comment (adipasquale) [reply app/models/agent.rb:30]: oh right I just tested I didn't know that; sorry.

slightly unrelated / larger than your PR but I think it'd be nice to add this to the Service model:

```
def self.secretariat
    find_by!(name: SECRETARIAT)
end
```

so we could rewrite these agents' scopes as: 

```
scope :secretariat, -> { where(service: Service.secretariat) } 
scope :available_for_rdv, lambda { |motif| 
  mot.for_secretariat ? where(service: [motif.service, Service.secretariat]) : where(service: motif.service)
}
```

**Comment by jjf21** [comment]
@adipasquale ok for you ?

**Pull request merged**